internal/tasks/core.go

```diff
@@ -80,15 +81,17 @@ func runNextTask() {
 		return
 	}
 
-	now := timestamp.Now()
-	if task.NextRunTime > now {
+	startTimestamp := timestamp.Now()
+	if task.NextRunTime > startTimestamp {
 		return
 	}
 
 	log.Debug("Task started.", task)
 	runTask(task)
 	log.Debug("Task finished.", task)
 
+	stats.AsyncStoreCourseTaskTime(startTimestamp, timestamp.Now(), task.CourseID, task.AssignmentID, task.UserEmail, task.Name)
```

Comment:
You should use the task's type instead of name.
The type will always exist, but the name can be empty.